DirectX11Proj/IScene.h

```diff
@@ -1,19 +1,37 @@
 #pragma once
 
+#include <list>
+#include <memory>
 #include <vector>
 
+
+#include "cameraclass.h"
+#include "inputclass.h"
+
 class IObject;
 
 class IScene
 {
 public:
+	friend class IApplication;
 	IScene();
 	~IScene();
 
-	virtual void Tick() = 0;
+	virtual void Tick(InputClass* const aInput) = 0;
 	virtual void Init() = 0;
 	virtual void Destroy() = 0;
 
-	std::vector<IObject> mObjects;
+	// Just a bit lazy and turn it into a vector, will convert back to a list later
+	std::vector <std::shared_ptr<IObject>> mObjects;
```

Comment:
I know you commented it, but do it now so you don't have to do it later
also, consider using unique_ptr if it's the only class managing its lifetime